### PR TITLE
feat(Snapchat): add presence

### PR DESCRIPTION
**Pull request by N0chteil**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

<details>
<summary>Settings</summary>

Screenshot_173
Screenshot_174
</details>

</details>

### CODE REVIEW

File: websites/S/Snapchat/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+			"icon": "fad fa-user-astronaut",
+			"value": true
+		},
+		{
+			"id": "showSnapStreak",
+			"title": "Show active chat SnapStreak",
+			"icon": "fad fa-fire-flame",
+			"value": true
+		}
+	]
```
Comment (EncryptedDev) [inline websites/S/Snapchat/dist/metadata.json:40]: I don't full agree with these settings, instead I think having the following settings would be more useful
- Privacy Mode - Hides the active chat username and active chat streak, but still displays the acttive chat Bitmoji
- Display Bitmoji - Toggle for showing the active chat's Bitmoji

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/S/Snapchat/dist/metadata.json (hunk 1 above)
Comment (N0chteil) [reply websites/S/Snapchat/dist/metadata.json:40]: But why not give the user full control?

Comment (EncryptedDev) [reply websites/S/Snapchat/dist/metadata.json:40]: If a user doesn't want to display the user's name then it is highly likely that they don't want to display the streak

File: websites/S/Snapchat/presence.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+		if (privacyMode && snapStreak) presenceData.state += ` | ${snapStreak}`;
+
+		if (showActiveChatAvatar && avatar) {
+			presenceData.largeImageKey = avatar;
+			presenceData.smallImageKey = "logo";
+		}
+	}
+
+	if (presenceData.details && window.location.pathname.split("/")[1])
```
Comment (DarkVillager) [inline websites/S/Snapchat/presence.ts:40]: Why the `window.location.pathname.split("/")[1])`

Comment (N0chteil) [reply websites/S/Snapchat/presence.ts:40]: To check if a chat is open

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Comment by EncryptedDev** [comment]
Thank you for contributing to the store! 🚀

**Pull request merged**